### PR TITLE
[ES-642] added config for sunbird-rc and insurance-credential.json templete

### PR DESCRIPTION
**Pull request by kaifk468**

### CODE REVIEW

File: esignet-mock.properties [hunk 1]
```diff
@@ -451,4 +457,15 @@ mosip.esignet.vci.key-values={ 'credential_issuer': '${mosip.esignet.domain.url}
...
-logging.level.io.mosip.esignet=DEBUG
\ No newline at end of file
+logging.level.io.mosip.esignet=DEBUG
+
+
+##---------------------------------sunbird-RC configuration------------------------------------------------------
+
+mosip.esignet.vciplugin.sunbird-rc.issue-credential-url=http://10.3.148.107/credential/credentials/issue
+mosip.esignet.vciplugin.sunbird-rc.supported-credential-types=InsuranceCredential
+mosip.esignet.vciplugin.sunbird-rc.credential-type.InsuranceCredential.static-value-map.issuerId=did:upai:678b6e24-9ff5-44ba-987e-53e458ccd1af
```
Comment (vishwa-vyom) [inline esignet-mock.properties]: can we update value to did:web:vishwa-vyom.github.io:test-project for now ?

File: esignet-mock.properties [hunk 2]
```diff
@@ -375,6 +375,11 @@ mosip.esignet.ui.wallet.config={{'wallet.name': 'Inji Mobile App', 'wallet.logo-
  'wallet.deep-link-uri': 'inji://landing-page-name?linkCode=LINK_CODE&linkExpireDateTime=LINK_EXPIRE_DT' },{'wallet.name': 'Inji Mobile App1', 'wallet.logo-url': 'inji_logo.png', 'wallet.download-uri': '#', \
  'wallet.deep-link-uri': 'inji://landing-page-name?linkCode=LINK_CODE&linkExpireDateTime=LINK_EXPIRE_DT' }}
 
+mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.individual-id-field=policyNumber
+mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.field-details={{"id":"policyNumber", "type":"text", "format":""},{"id":"fullName", "type":"text", "format":""},{"id":"dob", "type":"date", "format":"dd/mm/yyyy"}}
+mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.registry-search-url=http://10.3.148.107/registry/api/v1/Insurance/search
+mosip.esignet.authenticator.sunbird-rc.kba.entity-id-field=osid
+
```
Comment (vishwa-vyom) [inline esignet-mock.properties:380]: move these properties also to sunbird RC configuration section ?

File: esignet-default.properties [hunk 3]
```diff
@@ -459,3 +461,13 @@ mosip.esignet.vci.key-values={ 'credential_issuer': '${mosip.esignet.domain.url}
 
 logging.level.org.springframework.web.client.RestTemplate=DEBUG
 logging.level.io.mosip.esignet=DEBUG
+
+##---------------------------------sunbird-RC configuration------------------------------------------------------
+
+mosip.esignet.vciplugin.sunbird-rc.issue-credential-url=http://10.3.148.107/credential/credentials/issue
+mosip.esignet.vciplugin.sunbird-rc.supported-credential-types=InsuranceCredential
+mosip.esignet.vciplugin.sunbird-rc.credential-type.InsuranceCredential.static-value-map.issuerId=did:upai:678b6e24-9ff5-44ba-987e-53e458ccd1af
```
Comment (vishwa-vyom) [inline esignet-default.properties:472]: Can you update this value to did:web:${mosip.esignet.host}:.well-known

File: esignet-default.properties [hunk 4]
```diff
@@ -389,8 +391,8 @@ mosip.esignet.ui.config.key-values={'sbi.env': 'Developer', 'sbi.timeout.DISC':
   'consent.screen.timeout-buffer-in-secs': 5, 'linked-transaction-expire-in-secs': 240, 'sbi.port.range': 4501-4600, \
   'sbi.bio.subtypes.iris': 'UNKNOWN', 'sbi.bio.subtypes.finger': 'UNKNOWN', 'wallet.qr-code-buffer-in-secs': 10, 'otp.length': 6, \
   'password.regex': '\\S*', 'wallet.config': ${mosip.esignet.ui.wallet.config},\
-  'auth.factor.kba.individual-id-field' : ${mosip.esignet.authenticator.default.auth-factor.kba.individual-id-field},\
-   'auth.factor.kba.field-details':${mosip.esignet.authenticator.default.auth-factor.kba.field-details} }
+  'auth.factor.kba.individual-id-field' : '${mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.individual-id-field}',\
+  'auth.factor.kba.field-details':${mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.field-details}}
```
Comment (vishwa-vyom) [inline esignet-default.properties]: Undo this change, let it point to default configs

File: esignet-default.properties [hunk 5]
```diff
@@ -377,8 +377,10 @@ mosip.esignet.ui.wallet.config={{'wallet.name': 'Inji Mobile App', 'wallet.logo-
  'wallet.deep-link-uri': 'inji://landing-page-name?linkCode=LINK_CODE&linkExpireDateTime=LINK_EXPIRE_DT' },{'wallet.name': 'Inji Mobile App1', 'wallet.logo-url': 'inji_logo.png', 'wallet.download-uri': '#', \
  'wallet.deep-link-uri': 'inji://landing-page-name?linkCode=LINK_CODE&linkExpireDateTime=LINK_EXPIRE_DT' }}
 
-mosip.esignet.authenticator.default.auth-factor.kba.field-details={{'id':'policyNumber', 'type':'text', 'format':''},{'id':'name', 'type':'text', 'format':''},{'id':'dob', 'type':'date', 'format':'dd/mm/yyyy'}}
-mosip.esignet.authenticator.default.auth-factor.kba.individual-id-field='policyNumber'
+mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.individual-id-field=policyNumber
+mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.field-details={{"id":"policyNumber", "type":"text", "format":""},{"id":"fullName", "type":"text", "format":""},{"id":"dob", "type":"date", "format":"dd/mm/yyyy"}}
+mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.registry-search-url=http://10.3.148.107/registry/api/v1/Insurance/search
+mosip.esignet.authenticator.sunbird-rc.kba.entity-id-field=osid
```
Comment (vishwa-vyom) [inline esignet-default.properties]: move all 4 sunbird rc configs to the sunbird RC section

File: esignet-default.properties [hunk 6]
```diff
@@ -377,8 +377,10 @@ mosip.esignet.ui.wallet.config={{'wallet.name': 'Inji Mobile App', 'wallet.logo-
  'wallet.deep-link-uri': 'inji://landing-page-name?linkCode=LINK_CODE&linkExpireDateTime=LINK_EXPIRE_DT' },{'wallet.name': 'Inji Mobile App1', 'wallet.logo-url': 'inji_logo.png', 'wallet.download-uri': '#', \
  'wallet.deep-link-uri': 'inji://landing-page-name?linkCode=LINK_CODE&linkExpireDateTime=LINK_EXPIRE_DT' }}
 
-mosip.esignet.authenticator.default.auth-factor.kba.field-details={{'id':'policyNumber', 'type':'text', 'format':''},{'id':'name', 'type':'text', 'format':''},{'id':'dob', 'type':'date', 'format':'dd/mm/yyyy'}}
-mosip.esignet.authenticator.default.auth-factor.kba.individual-id-field='policyNumber'
```
Comment (vishwa-vyom) [inline esignet-default.properties:381]: don't remove the above default configs, rather assign value as ${mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.individual-id-field} and ${mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.field-details} respectively

File: esignet-default.properties [hunk 7]
```diff
@@ -459,3 +461,13 @@ mosip.esignet.vci.key-values={ 'credential_issuer': '${mosip.esignet.domain.url}
 
 logging.level.org.springframework.web.client.RestTemplate=DEBUG
 logging.level.io.mosip.esignet=DEBUG
+
+##---------------------------------sunbird-RC configuration------------------------------------------------------
```
Comment (vishwa-vyom) [inline esignet-default.properties]: Rename this to Sunbird-RC Plugin Configurations

Comment (vishwa-vyom) [reply esignet-default.properties]: update the same in other file also

File: esignet-default.properties [hunk 8]
```diff
@@ -96,11 +96,11 @@ mosip.esignet.kafka.linked-auth-code.topic=esignet-consented
 
 mosip.esignet.integration.scan-base-package=io.mosip.authentication.esignet.integration,io.mosip.esignet.mock.integration
 mosip.esignet.integration.binding-validator=BindingValidatorServiceImpl
-mosip.esignet.integration.authenticator=IdaAuthenticatorImpl
+mosip.esignet.integration.authenticator=SunbirdRCAuthenticationService
 mosip.esignet.integration.key-binder=IdaKeyBinderImpl
 mosip.esignet.integration.audit-plugin=IdaAuditPluginImpl
 mosip.esignet.integration.captcha-validator=GoogleRecaptchaValidatorService
-mosip.esignet.integration.vci-plugin=IdaVCIssuancePluginImpl
+mosip.esignet.integration.vci-plugin=SunbirdRCVCIssuancePlugin
```
Comment (vishwa-vyom) [inline esignet-default.properties]: Undo the above 2 changes, we want sunbird RC to be used only in mock eSignet

File: esignet-mock.properties [hunk 9]
```diff
@@ -375,6 +375,11 @@ mosip.esignet.ui.wallet.config={{'wallet.name': 'Inji Mobile App', 'wallet.logo-
  'wallet.deep-link-uri': 'inji://landing-page-name?linkCode=LINK_CODE&linkExpireDateTime=LINK_EXPIRE_DT' },{'wallet.name': 'Inji Mobile App1', 'wallet.logo-url': 'inji_logo.png', 'wallet.download-uri': '#', \
  'wallet.deep-link-uri': 'inji://landing-page-name?linkCode=LINK_CODE&linkExpireDateTime=LINK_EXPIRE_DT' }}
```
Comment (vishwa-vyom) [inline esignet-mock.properties:377]: default config needs to come here like the eSignet-default.properties with values pointing to the sunbird rc config

File: esignet-mock.properties [hunk 10]
```diff
@@ -383,7 +388,8 @@ mosip.esignet.ui.config.key-values={'sbi.env': 'Developer', 'sbi.timeout.DISC':
   'auth.txnid.length' : '${mosip.esignet.auth-txn-id-length}', 'consent.screen.timeout-in-secs':${mosip.esignet.authentication-expire-in-secs}, \
   'consent.screen.timeout-buffer-in-secs': 5, 'linked-transaction-expire-in-secs': 240, 'sbi.port.range': 4501-4600, \
   'sbi.bio.subtypes.iris': 'UNKNOWN', 'sbi.bio.subtypes.finger': 'UNKNOWN', 'wallet.qr-code-buffer-in-secs': 10, 'otp.length': 6, \
-  'password.regex': '\\S*', 'wallet.config': ${mosip.esignet.ui.wallet.config} }
+  'password.regex': '\\S*', 'wallet.config': ${mosip.esignet.ui.wallet.config},'auth.factor.kba.individual-id-field' : '${mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.individual-id-field}',\
+   'auth.factor.kba.field-details':${mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.field-details} }
```
Comment (vishwa-vyom) [inline esignet-mock.properties]: value to be same as eSignet-default.properties

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: esignet-mock.properties (hunk 1 above)
Comment (kaifk468) [reply esignet-mock.properties]: done

File: esignet-mock.properties (hunk 2 above)
Comment (kaifk468) [reply esignet-mock.properties:380]: moved

File: esignet-default.properties (hunk 3 above)
Comment (kaifk468) [reply esignet-default.properties:472]: changed

File: esignet-default.properties (hunk 5 above)
Comment (kaifk468) [reply esignet-default.properties]: moved

File: esignet-default.properties (hunk 6 above)
Comment (kaifk468) [reply esignet-default.properties:381]: done

File: esignet-default.properties (hunk 7 above)
Comment (kaifk468) [reply esignet-default.properties]: changed the name

File: esignet-default.properties (hunk 8 above)
Comment (kaifk468) [reply esignet-default.properties]: done

File: esignet-mock.properties (hunk 10 above)
Comment (kaifk468) [reply esignet-mock.properties]: now use value as default properties

File: esignet-mock.properties (hunk 9 above)
Comment (kaifk468) [reply esignet-mock.properties:377]: done

File: esignet-default.properties (hunk 4 above)
Comment (kaifk468) [reply esignet-default.properties]: done

File: esignet-default.properties (hunk 8 above)
Comment (vishwa-vyom) [reply esignet-default.properties]: this  is not taken care. We should have IDA in eSignet-default.properties and Sunbird in eSignet-mock.properties.

Comment (kaifk468) [reply esignet-default.properties]: done the changes

**Review by vishwa-vyom — APPROVED** [review]
